### ISSUE TITLE
Global find and replace fails 

### ISSUE THREAD

**Issue by GegznaV** [rstudio/rstudio#6819]
### System details

    RStudio Edition : Desktop or Server 
    RStudio Version : 1.3.957-1
    OS Version      : Windows 10
    R Version       : 4.0.0

### Steps to reproduce the problem

1) Create a new RStudio project that contains documents with non-English letters e.g., text
```
Apie atsitiktinę ir sistemingąją imties (pa)klaidas rašoma [@Cekanavicius2006, p.14-15, sk. 2.3].
```
You may use this project:
[replacement-issue.zip](https://github.com/rstudio/rstudio/files/4588589/replacement-issue.zip)

I will demonstrate by using my original project.

2) Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>F</kbd> in order to replace from `Cekanavicius2006` to `Cekanavicius_stat_I`. Next, follow the print-sceen instructions.

3)
![image](https://user-images.githubusercontent.com/12725868/81208837-53365900-8fd8-11ea-833e-9f2e1ba0c94b.png)

4)
![image](https://user-images.githubusercontent.com/12725868/81209617-6eee2f00-8fd9-11ea-96f8-9c4d84098d29.png)

5)
![image](https://user-images.githubusercontent.com/12725868/81209963-dc01c480-8fd9-11ea-9b6e-70f52a92199f.png)

6)
![image](https://user-images.githubusercontent.com/12725868/81209996-e7ed8680-8fd9-11ea-9926-849e551ea880.png)

7)
![image](https://user-images.githubusercontent.com/12725868/81210241-474b9680-8fda-11ea-9606-2e9e75a056aa.png)

Screenshots from GitHub Desktop for Windows:
![image](https://user-images.githubusercontent.com/12725868/81210325-677b5580-8fda-11ea-9af8-b30d94b9c5d9.png)
![image](https://user-images.githubusercontent.com/12725868/81210349-6ea26380-8fda-11ea-8147-523ccf7d0c1b.png)
![image](https://user-images.githubusercontent.com/12725868/81210364-72ce8100-8fda-11ea-9276-c3970188cd59.png)

It seems that the issue might be related to non-English characters, that preceded the text to replace. In places with correct replacement, there were no non-English letters.

![image](https://user-images.githubusercontent.com/12725868/81210516-a6111000-8fda-11ea-9af1-6bfe9e1e56fe.png)

### Describe the problem in detail

Global Find and Replace is not reliable: it fails to replace correctly. It is dangerous as using it without version control may result in damaged documents.

### Describe the behavior you expected

Replacement must work correctly.

**Comment by jmcphers** [comment]
Promoting this to 1.3 since we've confirmed that this is effectively a data loss bug.

**Comment by dfalty** [comment]
Closing https://github.com/rstudio/rstudio/pull/6841

**Comment by R-Hannibal** [comment]
QA Notes:

The fix for this ticket involved a refactor of global replace code (thanks @melissa-barca for the info). Considering how far into 1.3 regression QA is, we may want to do some additional testing in this area as part of the verification of this fix.

**Comment by R-Hannibal** [comment]
Verified Fixed on: Windows 10, R Version 4.0.0, RStudio Version 1.3.958-2
Global Find and Replace is now functioning correctly in scenarios where the text to be replaced is preceded by special or non-english characters.
